service/controller/aws/v14/resource_set.go

```diff
@@ -377,6 +379,18 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 	return resourceSet, nil
 }
 
+func transformFunc(obj interface{}) (v1alpha1.ClusterGuestConfig, bool, error) {
+	cr, err := awskey.ToCustomObject(obj)
+	deleted := awskey.IsDeleted(cr)
```

Comment:
Seeing this I think I understand why you want to go with the boolean return parameter. Let's please not do this. This interface mixes different primitives and anticipates that somebody understands its magic internals or intentions. That is not a clear design. I would say that when different functionality is required, different functions should be configured, where one function does one thing. 

One idea to get rid of the whole thing at all, `IsDeleted` works against the object metadata. There is https://godoc.org/k8s.io/apimachinery/pkg/api/meta#Accessor. So we do not need to configure something specific, because we can work with something abstract.